mshv_vtl/tdx: Handle some APIC functionality in kernel

jaredwhitedev opened the pull request:
TDX VMs require APIC emulation, previously handled by user-space. The transitions to user-space negatively impact performance. Avoid this overhead by moving a subset of the APIC emulation into the kernel.

jaredwhitedev commented:
User-Mode PR: https://github.com/microsoft/openvmm/pull/124

chris-oo commented on drivers/hv/mshv_vtl_main.c:
```diff
@@ -982,11 +1022,334 @@ static void mshv_vtl_idle(void)
+	ext_field_code.field_code    = SECONDARY_VM_EXEC_CONTROL;
+	ext_field_code.field_size    = 2;  /* TDX_FIELD_SIZE_32_BIT */
+	ext_field_code.context_code  = 2;  /* TDX_CONTEXT_CODE_VP_SCOPE */
+	ext_field_code.class_code    = 36; /* L2_VM1 (VTL0) */
```
I wonder does this change when we support VTL1?

jaredwhitedev committed:
mshv_vtl/tdx: Handle some APIC functionality in kernel

TDX VMs require APIC emulation, previously handled by user-space.
The transitions to user-space negatively impact performance.
Avoid this overhead by moving a subset of the APIC emulation
into the kernel.

Signed-off-by: Jared White <jaredwhite@microsoft.com>

benhillis commented:
This was merged in the internal repo. Future changes will come from this repo, but closing this one since it's already taken care of.
